metadata-db-app/src/cmr/metadata_db/services/concept_service.clj

```diff
@@ -950,7 +950,17 @@
             ;; collection revision is deleted. Indexer will index the variables associated with
             ;; the collection through the collection delete event. Not the variable association
             ;; delete event.
-            (when-not skip-publication
+
+            ;; CMR-8712: After deleting a collection, searching for services, tools, and generic
+            ;; documents that are associated with the collection, still show the collection association.
+            ;; This is because skip-publication is set to true for all associations when a collection is
+            ;; deleted. Given the above comments regarding the need for the tag and variable associations
```

Comment:
```suggestion
            ;; deleted. The associations for tag and variables to collections should remain see above
```

Reply:
Thanks. But "should remain" is too vague, in my opinion. I would rather summarize the huge comment block above, i.e. variable and tag association delete event should skip the publication.